$OpenBSD$

Use the BROGUE_FILENAME_MAX macro to prevent potential array mismatches.

Index: src/platform/main.c
--- src/platform/main.c.orig
+++ src/platform/main.c
@@ -124,8 +124,8 @@ int main(int argc, char *argv[])
 
 		if(strcmp(argv[i], "-o") == 0 || strcmp(argv[i], "--open") == 0) {
 			if (i + 1 < argc) {
-				strncpy(rogue.nextGamePath, argv[i + 1], 4096);
-				rogue.nextGamePath[4095] = '\0';
+				strncpy(rogue.nextGamePath, argv[i + 1], BROGUE_FILENAME_MAX);
+				rogue.nextGamePath[BROGUE_FILENAME_MAX - 1] = '\0';
 				rogue.nextGame = NG_OPEN_GAME;
 
 				if (!endswith(rogue.nextGamePath, GAME_SUFFIX)) {
@@ -139,12 +139,12 @@ int main(int argc, char *argv[])
 
 		if(strcmp(argv[i], "-v") == 0 || strcmp(argv[i], "--view") == 0) {
 			if (i + 1 < argc) {
-				strncpy(rogue.nextGamePath, argv[i + 1], 4096);
-				rogue.nextGamePath[4095] = '\0';
+				strncpy(rogue.nextGamePath, argv[i + 1], BROGUE_FILENAME_MAX);
+				rogue.nextGamePath[BROGUE_FILENAME_MAX - 1] = '\0';
 				rogue.nextGame = NG_VIEW_RECORDING;
 
 				if (!endswith(rogue.nextGamePath, RECORDING_SUFFIX)) {
-					append(rogue.nextGamePath, RECORDING_SUFFIX, 4096);
+					append(rogue.nextGamePath, RECORDING_SUFFIX, BROGUE_FILENAME_MAX);
 				}
 
 				i++;
@@ -197,15 +197,15 @@ int main(int argc, char *argv[])
 
 		// maybe it ends with .broguesave or .broguerec, then?
 		if (endswith(argv[i], GAME_SUFFIX)) {
-			strncpy(rogue.nextGamePath, argv[i], 4096);
-			rogue.nextGamePath[4095] = '\0';
+			strncpy(rogue.nextGamePath, argv[i], BROGUE_FILENAME_MAX);
+			rogue.nextGamePath[BROGUE_FILENAME_MAX - 1] = '\0';
 			rogue.nextGame = NG_OPEN_GAME;
 			continue;
 		}
 
 		if (endswith(argv[i], RECORDING_SUFFIX)) {
-			strncpy(rogue.nextGamePath, argv[i], 4096);
-			rogue.nextGamePath[4095] = '\0';
+			strncpy(rogue.nextGamePath, argv[i], BROGUE_FILENAME_MAX);
+			rogue.nextGamePath[BROGUE_FILENAME_MAX - 1] = '\0';
 			rogue.nextGame = NG_VIEW_RECORDING;
 			continue;
 		}
